content/en/docs/concepts/security/api-server-bypass-risks.md

```diff
@@ -0,0 +1,146 @@
+data held in the cluster.
+
+Access to the etcd API is typically managed by client certificate authentication.
+Any certificate issued by a certificate authority that etcd trusts allows full access
+to the data stored inside etcd.
+
+Direct access to etcd is not subject to Kubernetes admission control and is not logged
+by Kubernetes audit logging. An attacker who has read access to the API server's
+etcd client certificate private key (or can create a new trusted client certificate) can gain
+cluster admin rights by accessing cluster secrets or modifying access rules. Even without
+elevating their Kubernetes RBAC privileges, an attacker who can modify etcd can retrieve any API object
+or create new workloads inside the cluster.
+
+Many Kubernetes providers configure 
+etcd to use mutual TLS (both client and server verify each other's certificate for authentication).
+There is no widely accepted implementation of authorization for the etcd API, although
+the feature exists. Since there is no authorization model, any certificate
+with client access to etcd can be used to gain full access to etcd. Typically, etcd client certificates
+that are only used for health checking can also grant full read and write access.
+
+### Mitigations {#etcd-api-mitigations}
+
+- Ensure that the certificate authority trusted by etcd is used only for the purposes of
+  authentication to that service.
+- Control access to the private key for the etcd server certificate, and to the API server's
+  client certificate and key.
+- Consider restricting access to the the etcd port at a network level, to only allow access
+  from specified and trusted IP address ranges.
+
+
+## Container runtime socket {#runtime-socket}
+
+On each node in a Kubernetes cluster, access to interact with containers is controlled 
+by the container runtime (or runtimes, if you have configured more than one). Typically,
+the container runtime exposes a Unix socket that the kubelet can access. An attacker with
+access to this socket can launch new containers or interact with running containers.
+
+At the cluster level, the impact of this access depends on whether the containers that
+run on the compromised node have access to Secrets or other confidential
+data that an attacker could use to escalate privileges to other worker nodes or to
+control plane components.
+
+### Mitigations {#runtime-socket-mitigations}
+
+- Ensure that you tightly control filesystem access to container runtime sockets. 
+   When possible, restrict this access to the `root` user.
+- Isolate the kubelet from other components running on the node, using
+   mechanisms such as Linux kernel namespaces.
+- Ensure that you restrict or forbid the use of [`hostPath` mounts](/docs/concepts/storage/volumes/#hostpath)
+   that include the container runtime socket, either directly or by mounting a parent
+   directory. Also `hostPath` mounts should be set as read-only to mitigate risks
```

Comment:
I'd strengthen this to `must`, since arbitrary writeable hostPath mounts can pretty easily be used to access directories outside their apparent paths

Reply:
Cool I'm happy with that, I'll change.

Reply:
> since arbitrary writeable `hostPath` mounts can pretty easily be used to access directories outside their apparent paths

We could hyperlink to https://kubernetes.io/docs/concepts/storage/volumes/#hostpath and put further details there (there is already a warning callout).

Reply:
Is a read-write `hostPath` mount of type **`File`** also a risk? I wonder if that distinction is worth making.

Reply:
> Is a read-write `hostPath` mount of type **`File`** also a risk? I wonder if that distinction is worth making.

I... don't know... I'd recommend asking folks from sig-storage

Reply:
This might actually make a good topic for a separate page. There was a long discussion about the risks of `hostPath` mounts on [slack](https://kubernetes.slack.com/archives/C8P1DRTJA/p1661905584043699) and I don't think any of us had the whole picture on what exactly the risks are and how can attackers break out (past the obvious bit about vulnerabilities in things that manage host path mounts and the risks of local users getting using files in hostpath mounts for privesc)

Reply:
@raesene would you be willing to file an issue on this?

Reply:
yep can do :) I can file it in SIG-Security's repo, might be the best plce to start.